### PR TITLE
Beginners guide doc

### PR DESCRIPTION
**Pull request by Dgadavin**
## Description/Comments
A few sentences describing the overall goals of the pull request's commits / links to relevant github issues.

To consider:

* Is this feature suitable for gcdt or better a gcdt-plugin
* Does this feature have test coverage? (see docs for info about tests)
* Backwards compatibility

## Related PRs
Are there any other PR's on which this depends?

External PR | [title](https://github.com/glomex/gcdt/)

## Checklist before opening PR 
- [ ] Issue / PR description provided
- [ ] This feature has test coverage
- [ ] This feature has documentation (docs)
- [ ] The changelog has been updated (docs/CHANGELOG.md)

## Contributor License Agreement

gcdt is an open-source project that is licensed very permissively (see LICENSE). This means that software, and other contributions to the project are released to the public to be freely used, modified, remixed, and even sold without anyone’s permission. In order to protect this freedom, contributors must agree to waive their rights to their contributions BEFORE contributing them.

By submitting a pull request to gcdt you agree to this Contributor License Agreement, you are waiving your rights to the intellectual property that you contribute to the gcdt project. You agree to put your contribution under MIT license.

### CODE REVIEW

**Comment by markfink** [comment]
would be good to have quickstart guide in sync with the docu we already have (I think it would be confusing to have it explained in three places in different ways, my emphasis here is on using the `requirements_gcdt.txt` file as explained here):
https://github.com/glomex/gcdt#defining-which-gcdt-plugins-to-use
you plugin session does not reflect the discussion we had this morning. 1) gcdt-lookup does not exist. glomex-checks is a private glomex plugin. The docu for the private plugins has to be moved to our internal glomex SRE docs like we discussed in the morning.

**Comment by Dgadavin** [comment]
Should we have two userguide, one for internal users and one for external? Because the plugins is different and for glomex-* plugins users should have pip.conf with our python repo.

File: docs/03_getting_started.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    "VPCId": "lookup:stack:<stack-name>:DefaultVPCId",
+    "ScaleMinCapacity": "1",
+    "ScaleMaxCapacity": "1",
+    "InstanceType": "t2.micro",
+    "DefaultInstancePolicyARN": "lookup:stack:<stack-name>:DefaultInstancePolicyARN",
+    "AMI": "lookup:baseami"
+  }
+}
+```
+In `VPCId` and `DefaultInstancePolicyARN` used `gcdt-lookups` plugin which will search outputs in CloudFormation stack, that you mention in config file above, values for `DefaultVPCId` and `DefaultInstancePolicyARN` and use it in your template. *Instead of `<stack-name>` you should provide your stack name or use hardcoded value(not recommended).*
```
Comment (c0un7-z3r0) [inline docs/03_getting_started.md]: I would rephrase it to something like
"The values for `VPCId` and `DefaultInstancePolicyARN` are filled by by the `gcdt-lookups`. This plugin will search the outputs in the CloudFormation stack (as mentioned in the config) and use the values in your template. [...]"

File: docs/03_getting_started.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+```
+### Install gcdt and gcdt plugins
+#### Install gcdt
+First of all you need to create virtualenv and activate it. It's pretty easy.
+```bash
+$ virtualenv venv
+$ source ./venv/bin/activate
+$ pip install pip --upgrade
+```
+gcdt needs some gcdt-glugins so you should want to install these together. Plugins is a powerful tool to add features to `gcdt` without having to directly modify the `gcdt` core. The easiest way is to put the dependencies into a requirements_gcdt.txt file:
```
Comment (c0un7-z3r0) [inline docs/03_getting_started.md]: gcdt needs some gcdt-glugins so you should ~~want to~~ install these together

File: docs/03_getting_started.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    "DefaultInstancePolicyARN": "lookup:stack:<stack-name>:DefaultInstancePolicyARN",
+    "AMI": "lookup:baseami"
+  }
+}
+```
+The values for `VPCId` and `DefaultInstancePolicyARN` are filled by by the `gcdt-lookups` which then will be used in the template. The `gcdt-lookups` plugin will search the outputs in the CloudFormation stack (as mentioned in the config).
+
+*Instead of `<stack-name>` you should provide your stack name or use hardcoded value(not recommended).*
+It's time to create our first Infrastructure as Code. Let's do this.
+Here is simple [cloudformation.py](https://github.com/glomex/gcdt-sample-stack/blob/master/infrastructure/cloudformation.py) script. Use it as a template for creating your infrastructure.
```
Comment (jbrandstetter) [inline docs/03_getting_started.md]: Here is a simple...

File: docs/03_getting_started.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+### Deploy stack to AWS
+Before running deployment we need to set some necessary ENV variables. **Remember**: You need this ENV variables exported each time before running any `gcdt` command.
```
Comment (jbrandstetter) [inline docs/03_getting_started.md]: Before running a deployment...

File: docs/03_getting_started.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+```bash
+$ export ENV=dev
+$ export AWS_DEFAULT_PROFILE=glomex # Default profile. Generated by aws-mfa
+$ export AWS_DEFAULT_REGION=eu-west-1
+```
+Run your first infrastructure deployment. It's really easy.
```
Comment (jbrandstetter) [inline docs/03_getting_started.md]: It's really easy:

File: docs/03_getting_started.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+```bash
+$ kumo deploy
+```
+![Kumo deploy output](_static/gifs/kumo_deploy_output.gif "Kumo_Deploy_Output")
+**More information about `kumo` you can find in [docs](http://gcdt.readthedocs.io/en/latest/20_kumo.html)**
```
Comment (jbrandstetter) [inline docs/03_getting_started.md]: ... about kumo can be found in ...

File: docs/03_getting_started.md [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+
+## Ramuda
+Ramuda will help you to deploy, manage and control AWS Lambda. Ramuda supported runtimes are: **nodejs4.3, nodejs6.10, python2.7, python3.6**
```
Comment (jbrandstetter) [inline docs/03_getting_started.md]: ... Runtimes supported by ramuda are:

File: docs/03_getting_started.md [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+### Deploy simple AWS Lambda
+Create `gcdt_(dev|stage|prod)` file or update if you create it with `kumo` *(please change all values according your AWS account)*:
```
Comment (jbrandstetter) [inline docs/03_getting_started.md]: Create a ... file or update it if you created it with ...

File: docs/03_getting_started.md [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+            ],
+        }
+  }
+}
+```
+then do:
+```bash
+$ ramuda deploy
+```
+**More information about `ramuda` you can find in [docs](http://gcdt.readthedocs.io/en/latest/40_ramuda.html)**
```
Comment (jbrandstetter) [inline docs/03_getting_started.md]: .... ramuda can be found in ...

File: docs/03_getting_started.md [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+
+## Tenkai
+tenkai will help you to deploy your application using AWS [Codedeploy](https://aws.amazon.com/codedeploy/).
+tenkai will create bundle and upload it to s3 with all files that you have in `codedeploy` folder. Create or update `gcdt_(dev|stage|prod)` file *(please change all values according your AWS account)*:
```
Comment (jbrandstetter) [inline docs/03_getting_started.md]: ... will create an application bundle file ... have in your codedeploy folder. Create or update the ....

File: docs/03_getting_started.md [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+    "deploymentConfigName": "lookup:stack:gcdt-sample-stack:DeploymentConfig",
+    "artifactsBucket": "lookup:stack:<stack-name>:s3DeploymentBucket"
+  }
+}
+```
+then do:
+```bash
+$ tenkai deploy
+```
+**More information about `tenkai` you can find in [docs](http://gcdt.readthedocs.io/en/latest/30_tenkai.html)**
```
Comment (jbrandstetter) [inline docs/03_getting_started.md]: ... can be found in ...

File: docs/03_getting_started.md [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+
+## Yugen
+yugen is a tool that will help you to deploy and manage your API with AWS API Gateway. All you need is top put your `swagger.yml` into the same folder as a `gcdt_(dev|stage|prod)` file. Also, add some new configs into it *(please change all values according your AWS account)*:
```
Comment (jbrandstetter) [inline docs/03_getting_started.md]: ... you need is to put ...

File: docs/03_getting_started.md [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+"yugen": {
+    "api": {
+        "apiKey": "xxxxxxxxxxxxxx",
+        "description": "Gcdt sample API based on dp api-mock",
+        "name": "jenkins-gcdt-sample-api-dev",
+        "targetStage": "mock"
+    }
+}
+```
+**More information about `yugen` you can find in [docs](http://gcdt.readthedocs.io/en/latest/50_yugen.html)**
```
Comment (jbrandstetter) [inline docs/03_getting_started.md]: can be found in
